### PR TITLE
Remove hamming option from string distance tests

### PR DESCRIPTION
**Pull request by m3n3235**
Description: We should not test Hamming string distance for strings that are not equal length, since this is not defined. Removing hamming distance tests for unequal string distances.

### CODE REVIEW

**Comment by hinthornw** [comment]
The library by default pads strings to return a number. Is the recommendation that we should throw an error instead?

**Comment by m3n3235** [comment]
Somehow, it didn't pad strings for me. I got failed tests instead. Am I missing some config? This is what I get without this change:

======================================================================================================================================= short test summary info ========================================================================================================================================
FAILED tests/unit_tests/evaluation/string_distance/test_base.py::test_non_zero_distance[True-hamming] - ValueError: Sequences are not the same length.
FAILED tests/unit_tests/evaluation/string_distance/test_base.py::test_non_zero_distance[False-hamming] - ValueError: Sequences are not the same length.
FAILED tests/unit_tests/evaluation/string_distance/test_base.py::test_non_zero_distance_async[hamming] - ValueError: Sequences are not the same length.
FAILED tests/unit_tests/evaluation/string_distance/test_base.py::test_non_zero_distance_pairwise[hamming] - ValueError: Sequences are not the same length.
FAILED tests/unit_tests/evaluation/string_distance/test_base.py::test_non_zero_distance_pairwise_async[hamming] - ValueError: Sequences are not the same length.
====================================================================================================================== 5 failed, 1111 passed, 157 skipped, 26 warnings in 16.72s =======================================================================================================================
make: *** [Makefile:48: test] Error 1

File: libs/langchain/tests/unit_tests/evaluation/string_distance/test_base.py [hunk 1]
```diff
@@ -56,8 +56,13 @@ async def test_zero_distance_pairwise_async(distance: StringDistance) -> None:
     assert result["score"] == 0
 
 
+valid_distances = [
+    distance for distance in StringDistance if distance != StringDistance.HAMMING
+]
```
Comment (leo-gan) [inline libs/langchain/tests/unit_tests/evaluation/string_distance/test_base.py:61]: A comment about the reason for excluding Hummings would be nice.

**Review by leo-gan — APPROVED** [review]
LGTM wich a comment
@m3n3235 Thanks!

**Review by hinthornw — APPROVED** [review]

**Pull request merged**